tools/integration_tests/readonly/readonly_test.go

```diff
@@ -45,6 +50,49 @@ const ContentInFileInSubDirectoryTestBucket = "This is from directory Test/b fil
 const RenameFile = "rename.txt"
 const RenameDir = "rename"
 
+func createTestDataForReadOnlyTests() {
+	ctx := context.Background()
+	var cancel context.CancelFunc
+	// Define the text to write and the files to create
+	fileContents := []struct {
+		text string
+		file string
+		path string // Optional path within the bucket
+	}{
+		{"This is from directory Test file a", "a.txt", TestDirForReadOnlyTest + "/Test/"},
+		{"This is from file Test1", "Test1.txt", TestDirForReadOnlyTest + "/"},
+		{"This is from directory Test/b file b", "b.txt", TestDirForReadOnlyTest + "/Test/b/"},
+	}
+
+	// Create storage client before running tests.
+	ctx, cancel = context.WithTimeout(ctx, time.Minute*15)
+	client, err := client.CreateStorageClient(ctx)
```

Comment:
Nit: Can we keep this consistent by keeping the storage client common for the entire test package (in TestMain)?